Library/Homebrew/extend/object/deep_dup.rb

```diff
@@ -12,6 +13,7 @@ class Object
   #
   #   object.instance_variable_defined?(:@a) # => false
   #   dup.instance_variable_defined?(:@a)    # => true
+  sig { returns(T.self_type) }
   def deep_dup
```

Comment:
As @Bo98 mentioned: I think we can drop this.

Reply:
Naïvely swapping out `deep_dup` for `dup` in `abstract_artifact` leads to test failures in `test/cask/cask_spec.rb` – could i trouble someone with more context to take a deeper (pun intended) look into this one?